sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -183,17 +183,19 @@ case class DropTableCommand(
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
     val catalog = sparkSession.sessionState.catalog
-    // If the command DROP VIEW is to drop a table or DROP TABLE is to drop a view
-    // issue an exception.
-    catalog.getTableMetadataOption(tableName).map(_.tableType match {
-      case CatalogTableType.VIEW if !isView =>
-        throw new AnalysisException(
-          "Cannot drop a view with DROP TABLE. Please use DROP VIEW instead")
-      case o if o != CatalogTableType.VIEW && isView =>
-        throw new AnalysisException(
-          s"Cannot drop a table with DROP VIEW. Please use DROP TABLE instead")
-      case _ =>
-    })
+    if (tableName.database.forall(catalog.databaseExists) && catalog.tableExists(tableName)) {
```

Comment:
then `tableName.database.forall(catalog.databaseExists)` will return false and we won't hit this brach. Yea `Option.forall` is not obvious enough, let me update it.